addons/project_mail_plugin/controllers/mail_plugin.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import logging
+
+from odoo.exceptions import UserError
+from odoo.http import request
+
+from odoo.addons.mail_plugin.controllers import mail_plugin
+
+_logger = logging.getLogger(__name__)
+
+
+class MailPluginController(mail_plugin.MailPluginController):
+
+    def _get_contact_data(self, partner):
+        """
+        Overrides the base module's get_contact_data method by Adding the "tasks" key within the initial contact
+        information dict loaded when opening an email on Outlook.
+        This is structured this way to enable the "project" feature on the Outlook side only if the Odoo version
+        supports it.
+        """
+        contact_values = super(MailPluginController, self)._get_contact_data(partner)
+        if not partner:
+            contact_values['tasks'] = []
+        else:
+            partner_tasks = request.env['project.task'].search(
+                [('partner_id', '=', partner.id)], offset=0, limit=5)
+
+            tasks_values = []
+
+            for task in partner_tasks:
+                task_values = {
+                    'task_id': task.id,
+                    'name': task.name,
+                }
+                try:
+                    task.project_id.check_access_rule('read')
+                    task_values['project_name'] = task.project_id.name
+                except UserError:
+                    pass # user doesn't have access to the project name, we simply don't include it in the response
```

Comment:
Shouldn't we do something like putting "Internal project" or stuff like that ? That way people know there is a project associated with it that they don't see due to ACLs. I agree this is a corner case that should not happen often, just proposing ... maybe check with LNA ?

Reply:
I think this has been discussed with LNA and AWA, and it has been decided to not include these tasks.